packages/expo-google-sign-in/src/GoogleSignIn.js

```diff
@@ -1,190 +1,183 @@
 // @flow
+import { UnavailabilityError } from 'expo-errors';
+import invariant from 'invariant';
 
-import AuthData from './AuthData';
-import Authentication from './Authentication';
-import Identity from './Identity';
-import User from './User';
 import ExpoGoogleSignIn from './ExpoGoogleSignIn';
+import GoogleUser from './GoogleUser';
 
-const { ERRORS, SCOPES, TYPES } = ExpoGoogleSignIn;
+import type { GoogleSignInOptions, GoogleSignInAuthResult } from './GoogleSignIn.types';
 
-export type GoogleSignInType = 'default' | 'games';
+export const { ERRORS, SCOPES, TYPES } = ExpoGoogleSignIn;
 
-export type GoogleSignInOptions = {
-  scopes: ?Array<string>,
-  webClientId: ?string,
-  hostedDomain: ?string,
-  accountName: ?string,
+const DEFAULT_SCOPES = [SCOPES.PROFILE, SCOPES.EMAIL];
 
-  // Android
-  signInType: ?GoogleSignInType,
-  isOfflineEnabled: ?boolean,
-  isPromptEnabled: ?boolean,
-  // iOS
-  clientId: ?string,
-  language: ?string,
-  openIdRealm: ?string,
-};
+const { Constants } = requireConstants();
 
-export type GoogleSignInAuthResultType = 'success' | 'cancel';
-
-export type GoogleSignInAuthResult = {
-  type: GoogleSignInAuthResultType,
-  user: ?User,
-};
+function requireConstants() {
+  try {
+    return require('expo-constants').default;
```

Comment:
I'm not pretty sure if this will work correctly if there is no `expo-constants` installed.
iirc the bundler is checking whether the required/imported packages are installed, so if you don't have this package then I guess the CLI will throw appropriate error.